src/Data/List/Compat.hs

```diff
@@ -70,4 +129,15 @@ uncons                  :: [a] -> Maybe (a, [a])
 uncons []               = Nothing
 uncons (x:xs)           = Just (x, xs)
 
+-- | A strictly accumulating version of 'scanl'
+{-# NOINLINE [1] scanl' #-}
+scanl'           :: (b -> a -> b) -> b -> [a] -> [b]
```

Comment:
In that case, it might be a good idea to put `scanl'` in the newly added section of `README.markdown`.